homeassistant/components/apsystems_local/config_flow.py

```diff
@@ -0,0 +1,55 @@
+"""The config_flow for APsystems local API integration."""
+
+from aiohttp import client_exceptions
+from APsystemsEZ1 import APsystemsEZ1M
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_DEVICE_ID, CONF_IP_ADDRESS, CONF_NAME
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN, LOGGER
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_IP_ADDRESS): str,
+    }
+)
+
+
+class APsystemsLocalAPIFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Config flow for Apsystems local."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self,
+        user_input: dict | None = None,
+    ) -> config_entries.ConfigFlowResult:
+        """Handle a flow initialized by the user."""
+        _errors = {}
+        session = async_get_clientsession(self.hass, False)
+
+        if user_input is not None:
+            try:
+                session = async_get_clientsession(self.hass, False)
+                api = APsystemsEZ1M(user_input[CONF_IP_ADDRESS], session=session)
+                device_info = await api.get_device_info()
+                user_input[CONF_DEVICE_ID] = device_info.deviceId
+                user_input[CONF_NAME] = (
+                    "Solar"  # Set default name, can be changed later if desired easily
+                )
```

Comment:
There is no need to store these in the config entry data. The unique ID is stored as unique ID on the config entry, and the name should be sourced from the config entry title. That way users can change either the config entry title or the device name to change the name.
```suggestion
```